Task3/MoneyManager/MoneyManager.Business/Services/CategoryService.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using MoneyManager.DataAccess.Models;
+using MoneyManager.DataAccess.Repositories;
+
+namespace MoneyManager.Business.Services
+{
+    public class CategoryService : BaseService<Category>
+    {
+        private readonly CategoryRepository _categoryRepository;
+
+        public CategoryService(CategoryRepository categoryRepository) : base(categoryRepository)
+        {
+            _categoryRepository = categoryRepository;
+        }
+
+        public List<Category> GetAllParentCategories(string categoryName)
+        {
+            var selectedCategory = _categoryRepository.GetCategories().First(category => category.Name == categoryName);
```

Comment:
You get all categories from the database and then get first which satisfies your condition which is bad for performance. Move this logic to the repository level and get just one category from the database.